qchem/tests/test_deriv_grad.py

```diff
@@ -0,0 +1,96 @@
+import numpy as np
+import pytest
+
+import pennylane as qml
+from pennylane import qchem
+
+import shutil
```

Comment:
https://docs.python.org/3/library/shutil.html

Here I use it to delete the temporary files created by the PySCF/Psi4 involved in building the Hamiltonian.

Reply:
I really don't like that these files are created :angry:

Reply:
rather than using `shutil.rmtree`, it would be a lot safer to use pytest's built-in fixtures for temporary directories/files:

https://docs.pytest.org/en/stable/tmpdir.html

Reply:
I tried using this fixture but it did not work :-(. Perhaps because this folder is not created explicitly in the unit test but only when ``H(x)`` is evaluated in the ``derivative`` function?

Reply:
> rather than using `shutil.rmtree`, it would be a lot safer to use pytest's built-in fixtures for temporary directories/files:
> 
> https://docs.pytest.org/en/stable/tmpdir.html

Done. Please, ignore my comment above. I had forgotten the "trick" to delete temporary files :-)
```python
def H(x):
    return qchem.molecular_hamiltonian(["H", "H"], x, outpath=tmpdir.strpath)[0]
```
no need to use ``shutil``

Reply:
💪